src/Components/Open/GitHubFileBrowser.jsx

```diff
@@ -0,0 +1,160 @@
+  const accessToken = useStore((state) => state.accessToken)
+  const orgNamesArrWithAt = orgNamesArr.map((orgName) => `@${orgName}`)
+  const orgName = orgNamesArr[selectedOrgName]
+  const repoName = repoNamesArr[selectedRepoName]
+  const fileName = filesArr[selectedFileName]
+
+  const selectOrg = async (org) => {
+    setSelectedOrgName(org)
+    let repos
+    if (orgNamesArr[org] === user.nickname) {
+      repos = await getUserRepositories(accessToken)
+    } else {
+      repos = await getRepositories(orgNamesArr[org], accessToken)
+    }
+    const repoNames = Object.keys(repos).map((key) => repos[key].name)
+    setRepoNamesArr(repoNames)
+    setCurrentPath('')
+    setFoldersArr([''])
+    setSelectedFolderName('')
+    setSelectedFileName('')
+    setSelectedRepoName('')
+  }
+  const selectRepo = async (repo) => {
+    setSelectedRepoName(repo)
+    const owner = orgNamesArr[selectedOrgName]
+    const {files, directories} = await getFilesAndFolders(repoNamesArr[repo], owner, '/', accessToken)
+
+    const fileNames = files.map((file) => file.name)
+    const directoryNames = directories.map((directory) => directory.name)
+
+    setFilesArr(fileNames)
+    const foldersArrWithSeparator = [
+      ...directoryNames,
+    ]
+    setFoldersArr([...foldersArrWithSeparator])
+    setCurrentPath('')
+    setSelectedFolderName('')
+    setSelectedFileName('')
+  }
+
+  const selectFolder = async (folderIndex) => {
+    const owner = orgNamesArr[selectedOrgName]
+
+    // Get the selected folder name using the index
+    const selectedFolderName_ = foldersArr[folderIndex]
+
+    let newPath
+    if (selectedFolderName_ === '[Parent Directory]') {
+      // Move one directory up
+      const pathSegments = currentPath.split('/').filter(Boolean)
+      pathSegments.pop()
+      newPath = pathSegments.join('/')
+    } else {
+      // Navigate into a subfolder or stay at the root
+      newPath = selectedFolderName_ === '/' ? '' : `${currentPath}/${selectedFolderName_}`.replace('//', '/')
+    }
+
+    setSelectedFolderName('none')
+
+    setCurrentPath(newPath)
+
+    const {files, directories} = await getFilesAndFolders(repoName, owner, newPath, accessToken)
+
+    const fileNames = files.map((file) => file.name)
+    const directoryNames = directories.map((directory) => directory.name)
+
+    // Adjust navigation options based on the current level
+    const navigationOptions = newPath ? ['[Parent Directory]', ...directoryNames] : [...directoryNames]
+
+    setFilesArr(fileNames)
+    const foldersArrWithSeparator = [
+      ...navigationOptions, // All the folders
+    ]
+    setFoldersArr(foldersArrWithSeparator)
+  }
+  const navigateToFile = () => {
+    if (filesArr[selectedFileName].includes('.ifc')) {
+      navigate({pathname: `/share/v/gh/${orgName}/${repoName}/main${currentPath}/${fileName}`})
+      setIsDialogDisplayed(false)
+    }
+  }
+  return (
+    <Stack data-testid={'GitHub_access_controls'}>
```

Comment:
https://github.com/bldrs-ai/Share/wiki/Dev:-Style#data-testid

and below

Reply:
done